### PR TITLE
[wip] converting tests from  nibabel/tests/

### PR DESCRIPTION
**Pull request by djarecka**
will be adding more tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -132,6 +132,11 @@ script:
           cd for_testing
           cp ../.coveragerc .
           nosetests --with-doctest --with-coverage --cover-package nibabel nibabel
+          pytest -v ../nibabel/tests/test_a*.py ../nibabel/tests/test_b*.py ../nibabel/tests/test_c*.py ../nibabel/tests/test_d*.py
+          pytest -v../nibabel/tests/test_e*.py ../nibabel/tests/test_f*.py ../nibabel/tests/test_h*.py
+          pytest -v../nibabel/tests/test_i*.py ../nibabel/tests/test_k*.py ../nibabel/tests/test_l*.py ../nibabel/tests/test_m*.py
+          pytest -v../nibabel/tests/test_n*.py ../nibabel/tests/test_o*.py ../nibabel/tests/test_p*.py ../nibabel/tests/test_q*.py
+          pytest -v ../nibabel/tests/test_w*.py
           pytest -v ../nibabel/tests/test_affines.py ../nibabel/tests/test_volumeutils.py
```
Comment (effigies) [inline .travis.yml]: To test the installed version of nibabel, we can use:

```
pytest --pyargs nibabel
```

See https://docs.pytest.org/en/latest/goodpractices.html#tests-as-part-of-application-code

File: nibabel/optpkg.py [hunk 2]
```diff
@@ -115,6 +115,7 @@ def optional_package(name, trip_msg=None, min_version=None):
                     % (name, name, exc))
     pkg = TripWire(trip_msg)
 
+    # TODO dj: no clue why is it needed...
```
Comment (effigies) [inline nibabel/optpkg.py]: This allows a module that, for instance, depends on `h5py` to skip its tests if `h5py` is not installed.

Comment (djarecka) [reply nibabel/optpkg.py]: but you don't want to move this to the tests? i was surprised that this is a part of the main code.
If it's better  to keep it here, I'll just change simply change it to the pytest function

Comment (effigies) [reply nibabel/optpkg.py]: This affects doctests, not just tests.

**Comment by effigies** [comment]
@djarecka I rebased your changes on the latest `master`. If you'd like to check https://github.com/nipy/nibabel/compare/maint/3.0.x...effigies:ci/pytest_nibabeltests, and you're happy with the changes, you can `reset` to that branch.

If you'd prefer, I can move the pytest branch to upstream, and we can make PRs against it while we move to pytest. Whatever you like.

**Comment by djarecka** [comment]
@effigies - sorry, have been planning to finish this sooner. I'll come back to this at the beginning of Feb, and hopefully will finish in one session.

**Comment by effigies** [comment]
@djarecka No worries. Just saw that nose is going to start failing on Python 3.9, so thinking of trying to sprint on it: https://twitter.com/effigies/status/1220097622276562945

Got a few responses, so I wanted to see what the status of this one was. I'll open an issue for general discussion.

**Pull request closed without merging**